pandas/core/computation/expressions.py

```diff
@@ -81,10 +81,14 @@ def _can_use_numexpr(op, op_str, a, b, dtype_check):
             dtypes = set()
             for o in [a, b]:
                 if hasattr(o, "dtypes"):
-                    s = o.dtypes.value_counts()
-                    if len(s) > 1:
-                        return False
-                    dtypes |= set(s.index.astype(str))
+                    # Series implements dtypes, check for dimension count
```

Comment:
This has changed as I realised this wasn't working exactly as the previous function. We still need to pull out the dtype of the series to make sure it is supported by numexpr. The behaviour before was simply skipping over this bit which isn't correct